prompt_toolkit/buffer.py

```diff
@@ -1398,7 +1405,31 @@ def open_in_editor(self, validate_and_handle: bool = False) -> 'asyncio.Task[Non
             raise EditReadOnlyBuffer()
 
         # Write to temporary file
-        descriptor, filename = tempfile.mkstemp(to_str(self.tempfile_suffix))
+        remove = None
```

Comment:
This line needs to be removed, according to mypy. Otherwise, ``remove`` will be ``Optional[str]``, and ``rmtree`` below doesn't like that.

Can you do this, and then squash all the commits into one? Then I'll merge it.